src/rdo.rs

```diff
@@ -217,33 +215,35 @@ pub fn rdo_mode_decision(
 
     // Find the best chroma prediction mode for the current luma prediction mode
     for &chroma_mode in mode_set_chroma {
-      let mut wr: &mut dyn Writer = &mut WriterCounter::new();
-      let tell = wr.tell_frac();
-
-      encode_block_a(seq, cw, wr, bsize, bo, skip);
-      encode_block_b(fi, fs, cw, wr, luma_mode, chroma_mode, bsize, bo, skip, seq.bit_depth);
-
-      let cost = wr.tell_frac() - tell;
-      let rd = compute_rd_cost(
-        fi,
-        fs,
-        w,
-        h,
-        w_uv,
-        h_uv,
-        bo,
-        cost,
-        seq.bit_depth
-      );
+      for &skip in &[false, true] {
```

Comment:
Yes, putting skip/non-skip inside the loop for prediction mode, we can avoid duplicated run of each prediction, and I've pointed out this to rzumer.
I think your method here works fine since it passed the Travis testing.
The previous skip RDO has been reverted mainly due to it introduced corruptions.